DigitalLearningSolutions.Data/Services/CustomPromptsService.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using DigitalLearningSolutions.Data.Models.CustomPrompts;
+
+    public interface ICustomPromptsService
+    {
+        public CentreCustomPrompts? GetCustomPromptsForCentreByCentreId(int? centreId);
+    }
+
+    public class CustomPromptsService : ICustomPromptsService
+    {
+        private readonly ICustomPromptsDataService customPromptsDataService;
+
+        public CustomPromptsService(ICustomPromptsDataService customPromptsDataService)
+        {
+            this.customPromptsDataService = customPromptsDataService;
+        }
+
+        public CentreCustomPrompts? GetCustomPromptsForCentreByCentreId(int? centreId)
+        {
+            if (centreId == null)
+            {
+                return null;
+            }
+
+            var result = customPromptsDataService.GetCentreCustomPromptsByCentreId(centreId.Value);
+
+            return new CentreCustomPrompts
+            {
+                CentreId = result.CentreId,
+                CustomField1 = PopulateCustomPrompt(result.CustomField1PromptId, result.CustomField1Prompt, result.CustomField1Options, result.CustomField1Mandatory),
+                CustomField2 = PopulateCustomPrompt(result.CustomField2PromptId, result.CustomField2Prompt, result.CustomField2Options, result.CustomField2Mandatory),
+                CustomField3 = PopulateCustomPrompt(result.CustomField3PromptId, result.CustomField3Prompt, result.CustomField3Options, result.CustomField3Mandatory),
+                CustomField4 = PopulateCustomPrompt(result.CustomField4PromptId, result.CustomField4Prompt, result.CustomField4Options, result.CustomField4Mandatory),
+                CustomField5 = PopulateCustomPrompt(result.CustomField5PromptId, result.CustomField5Prompt, result.CustomField5Options, result.CustomField5Mandatory),
+                CustomField6 = PopulateCustomPrompt(result.CustomField6PromptId, result.CustomField6Prompt, result.CustomField6Options, result.CustomField6Mandatory)
+            };
+        }
+
+        private CustomPrompt? PopulateCustomPrompt(int promptId, string prompt, string? options, bool mandatory)
```

Comment:
I think in theory the "prompt" can be null in the db (but there are currently no null values). I think this is a constraint issue that can hopefully be resolved in the future when we get to tidy up the db, but until then, could we just pass back null if the "prompt" is null (so just do an check of id != 0 && prompt != null).